iceoryx_posh/include/iceoryx_posh/internal/popo/wait_set.inl

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_POSH_POPO_WAIT_SET_INL
+#define IOX_POSH_POPO_WAIT_SET_INL
+
+namespace iox
+{
+namespace popo
+{
+template <typename T>
+inline cxx::expected<Trigger, WaitSetError>
+WaitSet::acquireTrigger(T* const origin,
+                        const cxx::ConstMethodCallback<bool>& triggerCallback,
+                        const cxx::MethodCallback<void, const Trigger&>& invalidationCallback,
+                        const uint64_t triggerId,
+                        const Trigger::Callback<T> callback) noexcept
+{
+    static_assert(!std::is_copy_constructible<T>::value && !std::is_copy_assignable<T>::value
+                      && !std::is_move_assignable<T>::value && !std::is_move_constructible<T>::value,
+                  "At the moment only non copyable and non movable origin types are supported! To implement this we "
+                  "have to notify the WaitSet when origin moves about the new pointer to origin. This could be done in "
+                  "a callback inside of Trigger.");
+
+    Trigger logicalEqualTrigger(origin,
+                                m_conditionVariableDataPtr,
+                                triggerCallback,
+                                cxx::MethodCallback<void, const Trigger&>(),
+                                0,
+                                Trigger::Callback<T>());
+
+    for (auto& currentTrigger : m_triggerVector)
```

Comment:
What's being checked here? Can we add a comment ?
Also, why does `logicalEqualTrigger` have a hardcoded `triggerId` of `0` ?

Reply:
The idea behind this is documented in a comment and the hardcoded `0` is a bug which is fixed and where we have a unit test now.